### PR TITLE
Notification handling

### PR DESCRIPTION
**Pull request by fahslaj**

### CODE REVIEW

File: org.code.toboggan.network/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -10,7 +10,7 @@ Require-Bundle: org.eclipse.core.runtime,
  org.code.toboggan.core;bundle-version="1.0.0",
  org.code.toboggan.clientcore;bundle-version="2.0.0",
  org.eclipse.core.resources,
- org.eclipse.wb.core.lib
+ org.eclipse.wb.core.lib;bundle-version="1.9.0"
```
Comment (loganga) [inline org.code.toboggan.network/META-INF/MANIFEST.MF]: I noticed Ben pushed a commit earlier to project-skeleton where he removed the version numbers for these plugins, so we may want to do the same here

File: org.code.toboggan.network/src/org/code/toboggan/network/notification/clientcorelisteners/file/FileMoveNotificationHandler.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	private Logger logger = LogManager.getLogger(FileMoveNotificationHandler.class);
+	
+	@Override
+	public void handleNotification(Notification notification) {
+		FileMoveNotification n = (FileMoveNotification) notification.getData();
+		logger.info("Received file move notification for " + notification.getResourceID());
+		Set<ICoreExtension> extensions = extMgr.getExtensions(APIExtensionIDs.FILE_MOVE_ID);
+		for (ICoreExtension e : extensions) {
+			IFileMoveNotificationExtension p = (IFileMoveNotificationExtension) e;
+			p.fileMoved(notification.getResourceID(), Paths.get(n.newPath)); // TODO: convert to absolute path
```
Comment (loganga) [inline org.code.toboggan.network/src/org/code/toboggan/network/notification/clientcorelisteners/file/FileMoveNotificationHandler.java]: Should this TODO be resolved?

File: org.code.toboggan.network/src/org/code/toboggan/network/notification/clientcorelisteners/file/FileRenameNotificationHandler.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	private Logger logger = LogManager.getLogger(FileRenameNotificationHandler.class);
+	
+	@Override
+	public void handleNotification(Notification notification) {
+		FileRenameNotification n = (FileRenameNotification) notification.getData();
+		logger.info("Received file rename notification for " + notification.getResourceID());
+		Set<ICoreExtension> extensions = extMgr.getExtensions(APIExtensionIDs.FILE_RENAME_ID);
+		for (ICoreExtension e : extensions) {
+			IFileRenameNotificationExtension p = (IFileRenameNotificationExtension) e;
+//			p.fileRenamed(notification.getResourceID(), , n.newName); TODO: calculate new path
```
Comment (loganga) [inline org.code.toboggan.network/src/org/code/toboggan/network/notification/clientcorelisteners/file/FileRenameNotificationHandler.java]: Same here, what about this TODO?

**Review by loganga — CHANGES_REQUESTED** [review]
Just some small changes related to some unaddressed TODOs and the Manifest file.

File: org.code.toboggan.network/META-INF/MANIFEST.MF (hunk 1 above)
Comment (fahslaj) [reply org.code.toboggan.network/META-INF/MANIFEST.MF]: I'll rebase on top of project-skeleton which should fix this. I was the one who pushed those changes from Ben's computer when I was setting it up for him.

File: org.code.toboggan.network/src/org/code/toboggan/network/utils/NetworkUtils.java [hunk 4]
```diff
@@ -28,4 +30,17 @@ public static String toStringRelativePath(Path projectLocation, Path fileLocatio
...
+	/**
+	 * Turns a relative path and a filename into an absolute path.
+	 * @param relativePathString
+	 * @param filename
+	 * @return The Path object of the absolute path.
+	 */
+	public static Path toAbsolutePathFromRelative(Path relativePathp, String filename) {
+		Path wsPath = ResourcesPlugin.getWorkspace().getRoot().getLocation().toFile().toPath();
+		Path relativePath = Paths.get(relativePathp.toString(), filename);
+		Path abs = Paths.get(wsPath.toUri().toString(), relativePath.toUri().toString());
```
Comment (loganga) [inline org.code.toboggan.network/src/org/code/toboggan/network/utils/NetworkUtils.java]: Why toUri?

File: org.code.toboggan.network/src/org/code/toboggan/network/utils/NetworkUtils.java [hunk 5]
```diff
@@ -28,4 +30,17 @@ public static String toStringRelativePath(Path projectLocation, Path fileLocatio
 		return stringProjectRelative;
 	}
 	
+	/**
+	 * Turns a relative path and a filename into an absolute path.
+	 * @param relativePathString
+	 * @param filename
+	 * @return The Path object of the absolute path.
+	 */
+	public static Path toAbsolutePathFromRelative(Path relativePathp, String filename) {
```
Comment (loganga) [inline org.code.toboggan.network/src/org/code/toboggan/network/utils/NetworkUtils.java]: Rename dis pls (relativePathp)

File: org.code.toboggan.network/src/org/code/toboggan/network/utils/NetworkUtils.java (hunk 4 above)
Comment (loganga) [inline org.code.toboggan.network/src/org/code/toboggan/network/utils/NetworkUtils.java]: Why toURI()?

**Review by loganga — CHANGES_REQUESTED** [review]
Nit-picky things with the util function, but important nonetheless.

**Review by loganga — APPROVED** [review]

**Pull request merged**